signature/mechanism.go

```diff
@@ -28,6 +34,9 @@ type SigningMechanism interface {
 	// Sign creates a (non-detached) signature of input using keyIdentity.
 	// Fails with a SigningNotSupportedError if the mechanism does not support signing.
 	Sign(input []byte, keyIdentity string) ([]byte, error)
+	// Sign creates a (non-detached) signature of input using keyIdentity.
+	// Fails with a SigningNotSupportedError if the mechanism does not support signing.
+	SignWithOptions(input []byte, keyIdentity string, options *SignOptions) ([]byte, error)
```

Comment:
This is an API break.

The simplest way to handle this would be to provide a single passphrase to `NewGPGSigningMechanism`; that’s somewhat inaccurate (in that a single `SigningMechanism` can be used to `Sign` with different `keyIdentity` values, i.e. different passphrases, but not _too_ likely to be a problem in practice.

Alternatively, provide a `map[keyID]string`, or a `func(keyId)string`, as a parameter to `NewGPGSigningMechanism`. That would actually be better in that the passphrase callback is a per-GPGME-context state, so we can’t have two at a time, and concurrent calls of `Sign` with different `keyIdentity` values would be, _in principle_ possible — but it would also be very noticeably worse in that the `keyIdentity` is _supposed_ to be a key ID but in practice shorter values and email addresses work right now, and none of that would work with a generic callback AFAICT.

Or, well, go all the way, define a `SigningMechanism2` (and make it private), and all that …